src/store/filestore.rs

```diff
@@ -49,206 +43,196 @@ impl Store for LocalFileStore {
+        // Save the annotation file and throw and error if exist
         Self::save_file(
-            self.make_path::<T>(hash, SPEC_FILE_NAME),
-            &to_yaml::<T>(item)?,
+            self.make_path::<T>(
+                hash,
+                &Self::make_annotation_relpath(&annotation.name, &annotation.version),
+            ),
+            &serde_yaml::to_string(&annotation)?,
+            true,
+        )?;
+        // Save the pod and skip if it already exist, for the case of many annotation to a single pod
+        Self::save_file(
+            self.make_path::<T>(hash, Self::SPEC_RELPATH),
+            &to_yaml(model)?,
             false,
         )?;
 
-        // Save the annotation file and throw and error if exist
-        if let Some(value) = annotation {
-            // Annotation exist, thus save it
-            Self::save_file(
-                self.make_annotation_path::<T>(hash, &value.name, &value.version),
-                &serde_yaml::to_string(value)?,
-                true,
-            )?;
-        }
-
         Ok(())
     }
 
-    fn list_model<T>(&self) -> Result<Vec<ModelInfo>> {
-        let name_ver_tree = self.build_name_ver_tree::<T>()?;
-        let mut models = Vec::with_capacity(name_ver_tree.len());
-        for (key, hash) in name_ver_tree {
-            models.push(ModelInfo {
-                name: key.name,
-                version: key.version,
+    fn load_model<T: DeserializeOwned>(&self, model_id: &ModelID) -> Result<T> {
+        match model_id {
+            ModelID::Hash(hash) => from_yaml(
                 hash,
-            });
+                &fs::read_to_string(self.make_path::<T>(hash, Self::SPEC_RELPATH))?,
+                None,
+            ),
+            ModelID::Annotation(name, version) => {
+                let hash = self.lookup_hash::<T>(name, version)?;
+                from_yaml(
+                    &hash,
+                    &fs::read_to_string(self.make_path::<T>(&hash, Self::SPEC_RELPATH))?,
+                    Some(&fs::read_to_string(self.make_path::<T>(
+                        &hash,
+                        &Self::make_annotation_relpath(name, version),
+                    ))?),
+                )
+            }
         }
-
-        Ok(models)
     }
 
-    /// Generic function for loading spec.yaml into memory
-    fn load_model<T: DeserializeOwned>(&self, model_id: &ModelID) -> Result<T> {
-        match model_id {
-            ModelID::NameVer(name, version) => {
-                // Search the name-ver index
-                let hash = self.get_hash_from_name_ver_tree::<T>(name, version)?;
-
-                // Get the spec and annotation yaml
-                let spec_yaml = fs::read_to_string(self.make_path::<T>(&hash, SPEC_FILE_NAME))?;
-
-                let annotation_yaml =
-                    fs::read_to_string(self.make_annotation_path::<T>(&hash, name, version))?;
+    fn list_model<T>(&self) -> Result<BTreeMap<String, Vec<String>>> {
```

Comment:
Yeah we need to decide on what we want to send back for list

Reply:
Yes, we should discuss. Left some thoughts in this [comment](https://github.com/walkerlab/orcapod/pull/42#discussion_r1827683220).

Reply:
From our discussion, I've updated it to just pass through `Vec<ModelInfo>` for now. We can revisit once we are closer to the UI display rendering. Should be very easy to adjust structure to whatever we need.